workers/unity/Packages/com.improbable.gdk.core/Systems/WorldCommandsSendSystem.cs

```diff
@@ -75,67 +57,81 @@ protected override void OnUpdate()
                 return;
             }
 
-            for (var i = 0; i < createEntitySenderData.Length; i++)
-            {
-                var sender = createEntitySenderData.Senders[i];
-                var entity = createEntitySenderData.Entities[i];
-                foreach (var req in sender.RequestsToSend)
-                {
-                    var reqId = connection.SendCreateEntityRequest(req.Entity, req.EntityId, req.TimeoutMillis);
-                    createEntityStorage.CommandRequestsInFlight.Add(reqId.Id,
-                        new CommandRequestStore<WorldCommands.CreateEntity.Request>(entity, req, req.Context, req.RequestId));
-                }
+            var entityType = GetArchetypeChunkEntityType();
 
-                sender.RequestsToSend.Clear();
-            }
+            var createEntityType = GetArchetypeChunkComponentType<WorldCommands.CreateEntity.CommandSender>();
+            var deleteEntityType = GetArchetypeChunkComponentType<WorldCommands.DeleteEntity.CommandSender>();
+            var reserveEntityIdsType = GetArchetypeChunkComponentType<WorldCommands.ReserveEntityIds.CommandSender>();
+            var entityQueryType = GetArchetypeChunkComponentType<WorldCommands.EntityQuery.CommandSender>();
 
-            for (var i = 0; i < deleteEntitySenderData.Length; i++)
+            var chunkArray = group.CreateArchetypeChunkArray(Allocator.Temp);
+
+            foreach (var chunk in chunkArray)
             {
-                var sender = deleteEntitySenderData.Senders[i];
-                var entity = deleteEntitySenderData.Entities[i];
-                foreach (var req in sender.RequestsToSend)
+                var entityArray = chunk.GetNativeArray(entityType);
+
+                var createEntitySenders = chunk.GetNativeArray(createEntityType);
+                for (int i = 0; i < createEntitySenders.Length; ++i)
                 {
-                    var reqId = connection.SendDeleteEntityRequest(req.EntityId, req.TimeoutMillis);
-                    deleteEntityStorage.CommandRequestsInFlight.Add(reqId.Id,
-                        new CommandRequestStore<WorldCommands.DeleteEntity.Request>(entity, req, req.Context, req.RequestId));
+                    var sender = createEntitySenders[i];
+                    var entity = entityArray[i];
+                    foreach (var req in sender.RequestsToSend)
+                    {
+                        var reqId = connection.SendCreateEntityRequest(req.Entity, req.EntityId, req.TimeoutMillis);
+                        createEntityStorage.CommandRequestsInFlight.Add(reqId.Id,
+                            new CommandRequestStore<WorldCommands.CreateEntity.Request>(entity, req, req.Context, req.RequestId));
+                    }
+
+                    sender.RequestsToSend.Clear();
```

Comment:
I'm not sure what the overhead is for calling `Clear` on an already empty list

Reply:
This isn't a change from the previous system. So no worse than it previously was. It also doesn't appear in the profiler so I'm going to leave this for now.

Reply:
This wasnt changed because it was forgotten. This change was made in the normal command senders however
https://github.com/spatialos/gdk-for-unity/blob/master/workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt#L619